app/javascript/app/components/ghg-emissions/data-zoom/data-zoom-component.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Area, AreaChart, ResponsiveContainer } from 'recharts';
+import Draggable from 'react-draggable';
+import cx from 'classnames';
+import debounce from 'lodash/debounce';
+import throttle from 'lodash/throttle';
+import Icon from 'components/icon';
+import handleIcon from 'assets/icons/handle.svg';
+import styles from './data-zoom.scss';
+
+function DataZoom(props) {
+  const { data, onYearChange } = props;
+  const steps = data && data.length - 1;
+  const dataZoomRef = useRef();
+  const [width, setWidth] = useState(0);
+  const PADDING = 20;
+  const [position, setPosition] = useState({ min: 0, max: width - PADDING });
+
+  useEffect(() => {
+    if (dataZoomRef.current) {
+      const debouncedSetWidth = debounce(
+        () => setWidth(dataZoomRef.current.offsetWidth),
+        500
+      );
+
+      const refWidth = dataZoomRef.current.offsetWidth;
+      setWidth(refWidth);
+      setPosition({
+        ...position,
+        max: refWidth - PADDING
+      });
+
+      window.addEventListener('resize', debouncedSetWidth);
+      return () => {
+        window.removeEventListener('resize', debouncedSetWidth);
+      };
+    }
+    return undefined;
+  }, [dataZoomRef]);
+
+  const getYear = handleType => {
+    const handleStep = (steps * position[handleType]) / (width - PADDING);
+    const stepData = data && data[Math.floor(handleStep)];
+    return stepData && stepData.x;
+  };
+
+  const handleStop = () => {
+    onYearChange(getYear('min'), getYear('max'));
+  };
+
+  const handleDrag = (ui, handleType) =>
+    setPosition({
+      ...position,
+      [handleType]: position[handleType] + ui.deltaX
+    });
```

Comment:
Could you move all these functions and hooks to the container component? And leave only render methods here? Also please change the extension of this component file from `.jsx` to `.js`